{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "### header ###\n",
    "__author__ = \"Hunter Bennett\"\n",
    "__license__ = \"BSD\"\n",
    "__email__ = \"hunter.r.bennett@gmail.com\"\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "%matplotlib inline\n",
    "### imports ###\n",
    "import sys\n",
    "import os\n",
    "import re\n",
    "import glob\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats as stats\n",
    "from matplotlib_venn import venn3, venn2\n",
    "import seaborn as sns\n",
    "\n",
    "# import custom functions\n",
    "sys.path.insert(0, '/home/h1bennet/code/')\n",
    "from hbUtils import ngs_qc, quantile_normalize_df\n",
    "from homer_preprocessing import import_homer_diffgene\n",
    "from plotting_scripts import plot_exp_rpkm, gene_list_bar_plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "workingDirectory = '/home/h1bennet/strains_rna/results/A10_Manuscript_Figures_AllLiverCells//'\n",
    "leptinDirectory = '/home/h1bennet/strains/data/RNA/leptin_6h_strains_kupffer/'\n",
    "controlDirectory = '/home/h1bennet/strains/data/RNA/control_kupffer/'\n",
    "dataDirectory = '/home/h1bennet/strains_rna/results/K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped//'\n",
    "if not os.path.isdir(workingDirectory):\n",
    "    os.mkdir(workingDirectory)\n",
    "os.chdir(workingDirectory)\n",
    "\n",
    "if not os.path.isdir('./leptin_treatment'):\n",
    "    os.mkdir('./leptin_treatment')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "#### PLOTTING PARAMETERS FOR MANUSCRIPT ####\n",
    "# # get matplotlib to save readable fonts\n",
    "matplotlib.rcParams['pdf.fonttype'] = 42\n",
    "matplotlib.rcParams['ps.fonttype'] = 42\n",
    "matplotlib.rcParams['font.size'] = 6\n",
    "matplotlib.rcParams['savefig.dpi'] = 500\n",
    "\n",
    "\n",
    "# line widths\n",
    "matplotlib.rcParams['axes.linewidth'] = 1\n",
    "matplotlib.rcParams['xtick.major.width'] = 1\n",
    "matplotlib.rcParams['ytick.major.width'] = 1\n",
    "\n",
    "# adjust defualt color for plots to black\n",
    "# normal default is a dark gray\n",
    "COLOR = 'black'\n",
    "matplotlib.rcParams['text.color'] = COLOR\n",
    "matplotlib.rcParams['axes.labelcolor'] = COLOR\n",
    "matplotlib.rcParams['xtick.color'] = COLOR\n",
    "matplotlib.rcParams['ytick.color'] = COLOR\n",
    "matplotlib.rcParams['axes.edgecolor'] = COLOR\n",
    "\n",
    "#### PLOT PARAMETERS FOR THIS PLOT ####"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make hub to look at individual files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import differential gene files for examination and plotting"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make folder for figures"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.isdir('./figures/'):\n",
    "    os.mkdir('./figures')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Plot our own version of scatter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "comps = ['aj_kupffer_control.vs.aj_kupffer_leptin_6hours.scatter.homer.txt']\n",
    "\n",
    "ids = [['AJ.*Kupffer', 'aj.*leptin_6h']]\n",
    "\n",
    "labels = [['AJ control', 'AJ Leptin 6hours']]\n",
    "\n",
    "colors = [['#bdbdbd', '#a50f15']]\n",
    "\n",
    "# set plotting thresholds\n",
    "pthresh = 0.05\n",
    "fontsize=10\n",
    "labelsize=6\n",
    "lfcthresh = (np.log2(1.5),2)\n",
    "dotsize = 0.5\n",
    "ax_max = 16\n",
    "n_yticks = 5\n",
    "ytickstep = 3\n",
    "color = '#636363'\n",
    "filename='aj_leptin_treatment_combined_scatter.pdf'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "aj_kupffer_control.vs.aj_kupffer_leptin_6hours.scatter.homer.txt\n",
      "ge all transcripts (24940, 18)\n",
      "Diff Gene selected transcripts (9289, 14)\n",
      "ge selected transcripts (9289, 18)\n",
      "8018 peaks\n",
      "39 genes up in group 1\n",
      "27 genes up in group 2\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAJkAAACUCAYAAACAwotkAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAATJUlEQVR4nO2de3xV1ZXHvysvkkAkBiohCYGYBEISCQiWDmqVh09aHLF2qlUL1dZ2Wh/UsVqr1namtspYa1v9TDtjfY19WJFHW1ErUsVSR8XyiJY3MeGRYrwJhkByk5w1f5x74jXkPnK5576yv5/P/dzHOWefdZPf3XvttfdeW1QVg8FN0uJtgCH1MSIzuI4RmcF1jMgMrmNEZnAdIzKD62TE2wB/RERNSCWpkYE+NDWZwXWMyAyuY0RmcB0jMoPruOL4i8hE4DZgBTABGAnsVdWH3bifIbFxpSZT1e3Ao763rYACw924lyHxcb25VNXHVPV7QIaInNz/uIjcJSIqIiZ2kaKIG3EpESkEbgdygFeAIqAY+IaqeoNcZ+Jkyc2AcTJXRBYpRmRJjwnGGuKDEZnBdYzIDK5jRGZwHSMyAwCWZdHS0oJlWVEv24jMAIDH42HLli14PJ6ol21CGAbArsk8Hg8FBQWkpUVc95gQhmFgoiSwgBiRGVxtKsE0lwaiWpOZYSWD6xifzBAfjMiGAG7GwMIhpMhE5Fbf8/ki8rqIXO++WYZo4IirpaXFVcc+FOHUZOf4nj8PnA5c6Z45hmjgiOvgwYNs3ryZQ4cOUVNTQ0FBQVzsCWeOf7aIXAUcVNVuETka6oJ+c/x7gFpgInCtqnYfj8GG0DghiVGjRgHQ1NTEiSee6EoMLBxC9i5FZAZQBzwFdAMLVPWpkAWLnA3kq+oK3/ufAbeoakeQa0zv8jjp6emhoaEBVaWpqYnS0lJOOOEERo8eHQuRRdy7/A9VfVhV21W1MxyBHXNnkRuAVQMJzMzxjy6NjY00NjZy9Kjd4KhqrAQWkHBqsl8C24ANgAWgqi+FuMZ/jn8jUAX8H/CYqrYGuc7UZMeJ1+tl+/btFBQUsHXrVgCmTJnC6NGjY3H7yIKxIvKdfh+pb/VR1DEiiwz/iL3jj9XU1PQdj2FNZiL+qcrBgwepr69n3Lhx5OXlkZaWFq8mckCRhexdisha7MW5gt1D3K+qp0XXNkOkWJZFa6vtgTQ1NQFQW1sbVx+sPyFFpqqzndciMgz4uasWGQaFx+Nh3759AGRnZ9PZ2Rlni44lnJrMf9V3EXY4wxBHnGBrb28vHR0dTJo0CY/HQ2VlJYcPH45b0DUQ4Tj+j/heKtAGPKGqf3PFGOOThcSyLBoaGmhoaOj7rLS0lIqKivgZ9SGR+WSqulhExgJlwB5VPRBtywzh09LSQkNDA5mZmXR324MnI0aMiLNVwQmnubwZmANsBKaJyFpVvcd1ywzHYFkWbW1tAHR3d5OXl8f48eNjFQOLmHDGLheo6pnOGxF5FTAiiyFOHKy3t5e9e/cCtpNfV1dHVlZWnK0LTTj9XK+IzBKRHBE5E3v80hBDWlpa2Lx5Mx988EHfZ+Xl5UkhMAjP8R8H3AKUA7uAe1W10RVjjOP/EZxeZGtrK/v27aOqqoqMDLvxifd4ZAAij/iLyAlAvvPeiMw9HGE51NfX970eNWoUp5xySiKKyyHiiP8vgFJgv68QBb4YVdMMfXg8nj5hlZSUMHz4cDo67Mkr77//Ph6PJ+Ed/f6E01z+SVXPCXpStIwZ4jWZU4v19PSwb98+2tvb+46NHj2awsLCRG0mHQZXk4nIHN/LBt98sHrsWizkVB/D4LEsi127dtHU1ERWVhZe74dZTwsKCqiuru7zx5KNYD+JM32Pvdj+2Bm+92fEwK6UotfrZe/q5+n1Dpwu14niOwPc3n7nFRUVJa3AIHzHPw0YA7ynqj2uGZOizeXe1c+z7qrFnPn4I5RccN5Hjnm9XjZt2kR7ezsnnnhi34wKgMzMTMrLyyksLEzkJtKfwfUuReQBVb1BRD4N3AHswN744b9V9dGw7ypyPjATGAXcqqpHgpybkiLr9Xo5sGYtY+fOJt0X23Lm4jc3Nx9TczlUV1dTWFgYS1OPl0GL7EVVneebTzZfVY+ISDqwTlVnhX1XkQeAbwGLsXclWRnk3JQUmT9O9L6trY3GxoEjQRUVFeTm5rqWZcdFBr2QpEFEZgObgH8SkUxgGtAe5JqB+CmwBDiFAUYLhspCkl6vl6Znn2P3jh1s3ryZ3NzcgALKzs5O9F7koAj2Lb4GzAAqgQeBdcDlwBcGeY9hQBd2nO2F/gdV9S5VFVUd8FeQCvR6vWy55z5eveqLbF+xipycHLZu3Tpg2oDS0tKki4OFwszxjwFNzz7HuisXM/ziBYy49BIkc+Ce4vjx4ykrK0vmGswsJIkXB/fv580nnmTY1LqAAistLeXkk09OZoGBSR0VGwaKieWecALZp00fUGD5+flUV1engsACEs7YZRFwCXZAVgDcWneZCjT9cTXrr76WKfcvperzl3Hw4MG+Rbb9ycnJYerUqSkrLodwwsgrgR8Bm122JSVo3bQFLIudL7/Ce6UlBGv+p02blvICg/BE1qCqv3bdkiTHsixampuR0hJyF17EiEsuDigw7e5hjKeVzCEgMAjPJysQkQ0i8oTv8bjrViUhHo+HDf/7K965+VtkTZpIWk52wHOzdu1iy/VLOLBmbQwtjB/hTPUZ3/8zVX3XFWOSsHfpnyqztaWFPX9cHbQXOfakk8jeuZv09HSKzzunb5gpRRj0VJ8LVHU1MA/fFB8/fhlFw5ISR1yHDh3qmz0BkH3a9IDXVFVVYW3awrovfYUzH38k1QQWkGDNpROO7gF6/R6uzcJIJpp37WbdomtoeOfvYZ1fXFxMYWEhY+fO5szHH2Hs3NmhL0oRAtZkqvq872Wtqt7sfC4it7tuVQLjzKjY/evf0rX+NdpFyL/phqDXjBw5ksrKSrsnmZV1zHSfVCdgTSYiE0RkLjBPROb4HucCc2NnXmJhWRZbn1nBK1csIru4iOxZM8n78tVBr8nNzaWurm5IhCoCEeybj8eeBes/K3YG8O0Y2JWQeDweDuSNYNjM09j5i4fJPuuTpOcFThGQkZHBjBkzknpWazQI1ly+DLwsIncDo7FzYTSo6v5YGZdIWJZFT08PWQ3v0vXa6wxfeBHDpg6c4Ei7e+jauIk5X/vqkBcYhBfCuAs4FXshSS2wUVXvdMWYBAxhOD6YVE9m266dfQIKFqbofGMDh5beT92DD1B16SVDqamMOGfsy6p6lt/7V1T1k1E2zik74UTW+Idn+cuiazjp9luRKbVBz3UEmFVbQ27TXo6WTaBu+qkpNz8sCJEt7gXqReQy4G/YCfDqncR4qro75F1FyrAnOh4Gfq6qg51ZGze6Oztp2LGT3AXz0cqKgf+CPqyjnRx68L/oeu11Jtz9PbomVVI2fnzCJaSLB+GILBc41/dwuIPwV5J/GXgfyCJJkrVYlsXB/Qd45777aX70CRAhc9LEoIHWjuUr6Vr/GiddeB7Tr7qC9iNHknGOviuEmwSvGDuV55tATrAVRwOQA6wGSoD5wDL/gz6fr38a97hhWRa7d+9m+7LltD3+JLkXLyCrojygk+9Q9pUvsQeovf02huXmMiw3NzYGJwHh+GS3YTv8k7BDGM+q6gVh30BkKnZzmQn8QFX3BTk3Zj5Z/2VqvV4v+198iQ/GjuHdF14kq7YGb/3bQR1862gnHctXcs69PyA7P9/VfbyThIhnxp6jqpcDH/gUMKgBN1XdqKpLVPXrwQQWaw6sWcu6KxdTf+99dHd28vennmbdFYvY8dOHaLv3R3jr3w44m9WhY/lKOp5ezo6Hfh7P3PkJTzg+mdfXXKqIjMFeeZT0jJ07m+obr+PtH/+E9LIJ7HhxDVgWGaXjyP/mN0I2jwDTbrkZz/gJTL7xOvcNTmLCaS7Lge9jN5dbgW+H06uMyJgYhzCcJrO7s5O/Xn0tqJJ/8xKyP/HxgNc4YYrS88+lts5km+9HxNmvdwGf6ytF5DrsBbtJTa/XS9MfV9O6aQvWnE8y8t9uRIBh008NeI1293D4d8voWL6K2poaMCILi0jGPD5NkorM39k/sGYt66++FiyL4XubyLv8X4Je6y+wmiXXU3zuvBhZnfwMKS/1wJq1rLtqMftffImsulOYvPSH5C68iOEXXxTy2q6Nm+hYsYrqG6+n9ps3DZkJh9EgWMKVdRw7I1aAGlV1JYzttk/m1GRMrmL77l1hXeP4YB+b9QkK2zsomjfHCCwwg/PJ/HP3JwP+ez76hxEcYY2ZfRatHo+dsmnnDggj1NDXRD6zktrHf8m4T13o5ldIWVKmuXQ2E/V4PB/53Gkity9bzl/v/C7rF13DkQ2ht4bqbT/M+9/5dzqWraB6yQ3GBzsOUiYXRqCarLuzk+3LltO5bQfbfvYQwxdeFDTpiUPrf/6YrvWvMfoTH2fuqmdMExkeg16tNGAtp6rH5juKM/69Rn+BWZbFu+++y841L9HxzMqwBQYw9sbr6PnYx/j4/UuNwI6TYH/tNXzo+Pcp1JesLhN4WlUfcNG2sHGaxP45WT0eD9tXrKJjxe9BhIyK8rAElpeXx7Rp08g4NyaZ5VOeQTWXvtrtDFV9RUTWqGpUF5VE2lwOlJMV7JrsQNNeNj35KxTInn5qSJFVVlZSXFxsxiAjI+KZsZnYC3wXAGOBd1T1tqibR3RDGJZl0dzczLZt24ImPdHuHro2vIUCUy77HOPKJkTl/kOUQftkn8WO7ucCzwMTo11zuYX/xguh6NrwFm1L7wcRmDoVyia4bt9QI1ib8H3AC9ytqr8gwWdf+Cefa2lpCSkw7e6h840NWL5kdVVf/ypF8+YEvcYQGQFFpqqVwAPAp0VkFVAlIrNFZFjMrAuTXq+X+nvvY92Vi9n45K/7drcNRteGt2i75z5yWt5H0tIYfdoM04t0iaBesKpuxpf8zrd45BLgTmBQiRxEpAY4DzuT9h2q2hLikkGx7/k/8faPHmDUZZ+lrWgsh3y72w6EM0xk9fYCUHb2WUyaf+GQyk0Ra8LuQqnqblVdqqqD/m+o6tvAP4BC+i0miTSPv3/zqGqBQG9xUdDeozNM1HbPfaSlp1P30E8Yv+BTlFxwnqnFXCSmEX8RmQ80quqWAMfD7l3671ck1ZPZ9JvfBp2Pr909tP/2dxxZvoqRly5k+p23c1LRWBOqiC7xy34tIueLyDexwyDvRaNMJwXTmNlnsbf5QMj5+F0bN3Fkxe8BKDn9dApLTCwsViT92KWzCXwotLuH3D17GDEij6rPLCQzO3C6TUPERLyCPKHJz88PeU56ejoTJ09mzDnzTO0VB5JeZKHCFRkZGcyaNctk14kjSf+zLigooKSkJODx7OxsI7A4k/QiS0tLo6KigsrKSuDDSL5295CTk8OUKVPibKEh6UUGttBycnIAyN6zh0NL76ewtY2ZM2eSbRz8uJP0vUsHZ2bsyBEj+Mfal4+Z9mOICWYrQoPrmK0IDfHBiMzgOkZkBtcxIjO4jhGZwXWMyAyuY0RmcB0jMoPruD5yLCKnAguxl9bdoaodbt/TkFjEoia7DLgLWAGYdf9DkFjNgdF+z3303yxCJNjmMoYER1X1mH+g62OXIjId+Gfs5vI7qno4yLkDGnmc9zdlxrnMRBwgj/sfxZQZ3TJN79LgOokmsu+aMlOvzIRqLg2pSaLVZIYUJCGW8bgRsHUjyYsbuxCLyPnATGAUcOsg9xLtX9ZE4DbsmGQP9haSE4FrVTWiDW37lTkBGAnsVdWHwy0jUWqyqAdsgyV5OQ6+jC2waO5CfAGwFNjGcX53Vd0OPOp7/QdV/SFwhEFuHxmoTKAVO9Y5fDBlJIrIIEjANuICVZ8E/gcojVKRzi7EG7B3IY4GPwWWAKcQ5e2zReQGYFW0hvJU9TFV/R6Q4exDHw4J0VwCv8GuyXKJ0lbRvmZoClAerTKxf9HX4NuFOEplDsPOYrkfeOF4ChKRQuAzQI6I1AFV9sfyhqq2RqHMkdjbhBcDgZPA9S/D9C4NbpNIzaUhRTEiM7iOEZnBdYzIDK5jRBYmIrJTRPz3Yv+ziESldy4iZw8mJCAir0bjvrHCiCwMfOGAddg7tLjB2cAxIgu0U1+ykShxskRnIfAQcJuIDFPVAXdnEZGvYg87HQW+4nt+FDsWtkpV7/HNBC7BHqJpAP4VWARcLCIvAluwRwFGAN8WkfOwEzp3AYtUtdGVb+giKfFLiQGnquobwHPYm5kdg4icBFwKnO7b62AHcAv2bODTgTkiUuQ7/W1VnYc9EpGLLcSbVPUm3/E2VZ0PNANzfNffCXzLlW/nMkZkIRCRcqBWRJ4DPoddqwxEGfCWqvZC3+az5cBbvuN/850DUO973o894NyfDb7nCfh2hAHeBCoi+xbxxYgsNJcA16jq+b4aaqyIpA9w3m5gmuNH+Z53A9N9x6dhN4/w0fFZwR6z9C/T2R25AajzvZ4B7DqubxInjE8WmvnYg9gO7wBn+F73bZWtqu+JyDJgvYg4Ptk9wGMikgX8XlX3BViN9WfgbhGZCTT6ldksImtFZD32jn1fiN7Xih1m7DJCROQvPl/JEALTXEaAiLwELIu3HcmCqckMrmNqMoPrGJEZXMeIzOA6RmQG1zEiM7iOEZnBdf4fK11EUVhNBvIAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 144x144 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig, ax = plt.subplots(1,1,\n",
    "                        figsize=(2,2),\n",
    "                        sharex=True,\n",
    "                        sharey=True)\n",
    "plt.subplots_adjust(wspace=0.2,\n",
    "                    hspace=0.2)\n",
    "\n",
    "# set count for labeling axes\n",
    "count=0\n",
    "\n",
    "for comp, grep, labs, color in zip(comps, ids, labels, colors):\n",
    "    print(comp)\n",
    "    de, tpm, tpm_mat, tpm_mat_quant = import_homer_diffgene(\n",
    "        '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/'+comp,\n",
    "        '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/expression/HOMER.rawTPM.txt',\n",
    "        gene_name_index=True)\n",
    "    \n",
    "    # extract columns\n",
    "    g0 = tpm_mat.columns.str.contains(grep[0])\n",
    "    g1 = tpm_mat.columns.str.contains(grep[1])\n",
    "    col = g0 | g1\n",
    "    \n",
    "    de['logtpm'] = np.log2(tpm_mat.loc[:, col].mean(1)+1)\n",
    "    de['log10p'] = -np.log10(de.padj + 10**(-50))\n",
    "    de['g0_mean'] = np.log2(tpm_mat.loc[:, g0].mean(1)+1)\n",
    "    de['g1_mean'] = np.log2(tpm_mat.loc[:, g1].mean(1)+1)\n",
    "       \n",
    "    # plot group by group scatter:\n",
    "    de_nonsig = de.reindex(de.index[de.padj >= pthresh])\n",
    "    de_sig_fc1 = de.reindex(de.index[(de.padj < pthresh) & (np.abs(de.log2FoldChange) >= lfcthresh[0])])\n",
    "    \n",
    "    # check number of peaks in each group\n",
    "    print(de_nonsig.shape[0], 'peaks')\n",
    "    print((de_sig_fc1.log2FoldChange > lfcthresh[0]).sum(), 'genes up in group 1')\n",
    "    print((de_sig_fc1.log2FoldChange < -lfcthresh[0]).sum(), 'genes up in group 2')\n",
    "    \n",
    "    for df, c in zip([de_nonsig, de_sig_fc1],\n",
    "                     color):\n",
    "        ax.scatter(\n",
    "            x=df.g0_mean,\n",
    "            y=df.g1_mean,\n",
    "            c=c,\n",
    "            s=dotsize,\n",
    "            #rasterized=True\n",
    "        )\n",
    "     \n",
    "    # Hide the right and top spines\n",
    "    ax.spines['right'].set_visible(False)\n",
    "    ax.spines['top'].set_visible(False)\n",
    "    \n",
    "\n",
    "    ax.set_xlabel(labs[0], fontsize=labelsize+2)\n",
    "    ax.set_ylabel(labs[1], fontsize=labelsize+2)\n",
    "    \n",
    "    #set max and min\n",
    "    ax.set_xlim([0,ax_max])\n",
    "    ax.set_ylim([0,ax_max])\n",
    "    \n",
    "    # set ticks\n",
    "    ax.set_xticks(np.arange(0,ax_max,ytickstep))\n",
    "    ax.set_xticklabels(np.arange(0,ax_max,ytickstep),\n",
    "                       fontdict={'fontsize':labelsize})\n",
    "    ax.set_yticks(np.arange(0,ax_max,ytickstep))\n",
    "    ax.set_yticklabels(np.arange(0,ax_max,ytickstep),\n",
    "                       fontdict={'fontsize':labelsize})\n",
    "    \n",
    "    count = count + 1\n",
    "plt.savefig('./leptin_treatment/'+filename, bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Metascape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta = pd.read_excel('../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/gene_lists/aj_kupffer_leptin_6hours_up_kupffer_genes_fc_0.58_pval_0.05_/metascape_result.xlsx',\n",
    "                     sheet_name=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Mononuclear Cell\\nDifferentiation',\n",
       " 'AMPK\\nSignaling',\n",
       " 'Reg. of Cellular\\nCatabolism']"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "['GO:1903131', 'ko04152', 'GO:0031331']\n",
    "['Mononuclear Cell\\nDifferentiation',\n",
    "                      'AMPK\\nSignaling',\n",
    "                      'Reg. of Cellular\\nCatabolism']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "leptin_act_terms = ['R-MMU-1799339', 'GO:0006119', 'GO:0032642']\n",
    "leptin_act_strings = ['Antigen\\nPresentation',\n",
    "                    'Oxidative\\nPhosphorylation',\n",
    "                    'Chemokine\\nProduction']\n",
    "leptin_rep_terms = ['GO:1903131', 'ko04152', 'GO:0031331']\n",
    "leptin_rep_strings = ['Mononuclear Cell\\nDifferentiation',\n",
    "                      'AMPK\\nSignaling',\n",
    "                      'Reg. of Cellular\\nCatabolism']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAJ0AAADRCAYAAADWvLDBAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAUA0lEQVR4nO2dfbxVVZnHv7+LL6ApqJhmoYhOCkgfNNPMVIrGKW34YDOm46i9ahK+ICmGWqKJkPlCMr5gOaOihDommaJTQkgvam8ivsRYMmqiQ2IjagqCPvPHs453c7j3cu/lnLXPvff5fj73c87Ze+2914HfWWuvtZ/femRmBEFOmsquQNDzCNEF2QnRBdkJ0QXZCdEF2QnRBdkJ0W0EkkZIGtfBYyZJGp7eX97OY+ZKOnVDdZB0kqQ92nnOOe0oM1DStPacryNsUusT9mQkDQDOBAQ8b2ZTJC0BvgcMBSYAHwV2lHQrsGs6bi4wL5W51sweLJxzMPBr4GDgCkm9gKuA14ClwDbARyW9AuwI9JE0G/gcYMCNwDjgFGBbYAHwOLCnpEnA9FTn3unvNOCEVJdVNf9HIlq6WnMysBp4CdgrbVtuZpcCM4GjgV8A15jZ/MJxTcA0YApwRNU5T0zH/lbSgcA/AI+Y2dfM7Mp0vnvN7N8Lx9wFHA58Or1fC2wK/AX4nJk9Biwxs0nAfsBuwMtAL2AQ8EkzGwPcunH/HC0TLV1taQJmm9nvCts2rXp9u4XjVpnZW5LWAJtXNkrqDRwK9AHeBewB/GfVOVo63w+Ba/AW9yvAGOAe4OfAz1KZyqOoJuDB9MOoXHd1pV6tftONIES38RwpaU/gLeDbwFRJy4A1ZnY20FfSRcDuwFhgX2CCpOvbc27gYjO7AUDSbXhXO0rSxXj3Oied7x2BmNnrkt4ovP8VLrwPA2+mYs9LugT4DjA6vd8avwWYL+nc9LnmKJ691hdJc8xsdNn1aCRCdEF2YiARZCdEF2QnRBdkJ0QXZCemTBKSYkRVY8xMLW0P0RWIkXztkFrUGxDda1AC0dIVaOvXGbTNgAEDePbZZ9tVNkRXYN68eWVXocsycuTIdpeN7jXIToguyE6ILshOiC7ITkOKTtKpkia2sH1aet1O0ify1yyoBQ03epW0CR49+4qk0/C4/q2B7wNDJH0eD9HeSz7HcSDQH7gP/xHtiXsFzgNuBmYDg81sPREH5dCILd1oYLv0txr4MW4u+SDwhJldj8f8A4wCJgMVv8FngOeBZbjpZamZ3YiHewcNQsO1dMAhZnYKgKSXgY/h8fxNwLOSTgbuTWXvAs4G3o17AObgLd1fgWdpFmfQQDSc6CqCS+/7FXYtqio6TdIOwM5AX+AhM3uxqsy4dJ4WvanJgnfextY56Bh1614lfV7SxZKuljS0Budbz/RrZsuBoWY2HjhAUoe6UTObZGZqLRoiqA/1bulm4YOCxZKOBx4FjsMdSXfjXedKYDHupvoUfv91NXAR8CN8oHA5PogYkcrtAwwAZhS27wMslDQBWIEPQGYClwILgZXpfjAomXoPJI7BTb/XmtnNuIdzBW6d2x34DS7KrXAj8tPpb3fgL+mYN83sj/ggYgE+kv0b3sIVt1d4bzIhvy99fsDMvgsMr9/XDDpCvUU3y8y+iY9CwW/8dyC1fkC/tG8Y7iYfhLdkT7L+IGArSYfizvnXcTd6cXuFZZLGAs+lzzGYaDDCgpiQZBFl0nlGjhy5ThCspFYjhxtxni7o5oToguyE6ILsxD1dItxgG0d1uHpb93QN90SiTOIHmIfoXoPshOiC7IToguzEPV2B7up77YgnNQchugLd9YlERzypOYjuNchOiC7IToguyE6ILshOqaKTtJOkqZLGSbpQUlNh3zre1pQDa72l8Qte2FEdDVcPyqHs0etYYLKZvSrpMOAbkhYAh+Eh60Vv6/bATyUdROvh6r+TNMbMzpU0Hc97dQIpPN7MfpX36wUt0Wjd68O4r+LSwrZqb2tb4ep/A5okfQT4fTq2GB4fNABlt3RX4a3bC7i5egvgE8BE4IpU5i48xdCuwP14uPqfaT1cfVb62x/YBU/o9ipQzNcVlEiPDm2q9r1258nh3P/PEa7eCuF7LYcOi67KRP3Tzlw0ZXfut+GS6x23nuE6VnLqenT2nq5iop4vaTzNqyRNxdcQeQHPFToYeAr4X9wDu5hma+BJyfk/Pu0z3Kt6Fr4q03T8/m0K8AXgpwCStsEN23vgiXmrV3IyPHt0XzwV5nSSadvMvtrJ7xvUkM52rxUT9Y1mdhkuwGHAoylZ7SHAY8Aa3NsKbnr+N+CT6fN1wB24MIea2RX41EZ/YLGZzQb+A/gSMMjM/jsd1yfVezmev7R6JafRuPjn4inI3zFtd/K7BjWms6KrNlEbvmTEMElfw/PEvxvPM19JK36QpHOAn6TPa/CszE3AY5JOBTbDpzhWA5jZU3i67zsK1x5I88i1F80rOVVa7TnA1/G5voWE2brhyDJ6TZO3/cxsTgeP2w/4ZzObUJeKrXutbmu2brTRa5Z5uqq1Rjpy3K/x9OFBN6JHT5kE5dCjJ4eLdGffaxnh6qV3r12F+AHmIbrXIDshuiA7IbogO3FPV6A7+V4bzetaJERXoDtNDjea17VIdK9BdkJ0QXZCdEF2QnRBdrINJCS9B7ccvoTHtv3SzKrzfXXkfCMoRK5IOsrMbqlFXYP6knP0OhaYamavpUjfYyWNwV1fe7BuntabcMvhUNyW+H7gVOA7wDPpfI8CWyRzzSzgAOAWSQuBG/Bg0OuAc/AYvT+Y2Q/r/zWDDZGzexUe7FnhFtws/UHWz9P6dIpAXm1m0/Fgz+2BVelzJTD0DGCOmT1ZOO9iM7sOeA8wAg/iXEZz2qagZHK2dFcCEyVVutc1aXsT6+dprewrRia/CPROEcaPpu0XAp9O2bArFCOF78dD1rfEW8ygAejRoU3d2fdaRrRwkfC9tkL4Xsshu+g6knw4lW0zZWb4XrseZT17rU4+/AhwbNo3E9gXX9tkf2CRpGlmNq7wehHun32I8L12OcoS3TG4GftaM7s5ZaP+ftp3BLCbmZ2U7IxFeknaHnjNzK4CkPSEmV0vaWAqMxof1R5Ewfcq6UP1/UpBeynrnq7aNzsX+HL6uxt4UNLn8OkUgBcknQAMN7MXgS0ljZG0D+F77XL06NFrke7me43RaxAUCNEF2YnuNdHdfK9lh6uH77WdxA8wD9G9BtkJ0QXZie61QHeyIDYyIboCjfDgX5KVXY961yG61yA7IbogOyG6Zs5vb8GUHG9cR06e0hgMT+8v31A9JP1G0nRJt0v6QEeulY7fmO5xnX+Lznzftoh7uoSZTerssZIGAGfiPpDnzWyKpCXA93Bz0QQ83GpHSbfiPhAkzQXmpTLXmtmDhXosM7NTJH0c+BgeBjYRX0C8Hx5JcynwNB6O/wCeD+0HeCjXD5LxqXf6Ow23DLyGm5tm4h6VZ/CFyRcAF6TvsCbV78J07n54+FlNiJauNpyMR8y8RLNpaHkyF80Ejsbj/a4xs/mF45rwXBhT8JCuIu+VdBVwCe5yGwx8HFiJC6fS+t1gZmfSHI+4xMymAkOA3YCX8VXoB+HmpPvxH0PfdP0f4ZE9/4oL7P+AHSTtAuxqZqfTvCJ+TYiWrjY0AbPNrJj0btOq17dbOG6Vmb0laQ0e1FpkmZl9VdJRwKfwBb+XFFvkFP61aepKK/+XKwt1ejAJv1L+CDzHx51m9vEULvZpXHCPAPea2e2pbF+ac2+sas8/QnsJ0XWeIyXtCbyFRyhPlbQMWGNmZwN9U4Tz7rjnd19ggqTrO3idW/FMkLOBN9L94OZA5b7wi7g988aq4/4LGC3pEjxd6TnAt/D0pE9KGobnwt0Ez2Q0E7hK0oeBrVIQ7dOSzsSdeo9SI+KBf52QNMfM1svEXeNrXA+MM7OX63mdWhOiC7ITA4kgOyG6IDshuiA7IbogOzFlkuhu4eqNQISrt4MYydeOth79RvcaZCdaugI9IXK4bJcYhOjWoTs5/FujEZKaRPcaZCdEF2QnRBdkJ0QXZKdN0aXlV2el96dKuqaelaks5drBY4ZIOl/SeEnHtrB/WvE1KJ/2jF5XSBoEbAuskrQjcBYeBr0Qj0RdgYdPn4Ovsml4aPRZeHh0JZnIQ3i8/QJgHO4P+Bi+WuY1AJLejwcXvomHUfcC9gEGmNkZkn4JTDez2al+XwbOMLO30/EH4pG2fYCrO/WvEtSV9nSvs4Gr8EhU8CVV7wam0hzXPxu4BxgMDDWzK3DR9GfdZCKGGz96peOOMLMLKoJLjMJFvBSPut0aj3atLIq9uCC4ljgaN6s8nY4PGoz2iO514HQzeyB9Xggcji+xekfathYXVBPwWEowshkunuLyq0uAQ4Hj0+c7JH0zxepXuAvYgbQQNm50eZ1moa5mXa4Dzit0r7fiJpS3gCcJGo4eHTncnZOXtEauZWHruvxrIS/EeEkTknsJSTMkHSTpEkmHd/LcAysm38p5W7j28Nb2b4hIXlIOtXoMNsvMFiVz75ckPYkbil/BBxi3pWX738bzQ8wAvov7Lzdj3QyIN+P3iIPxTIj7SdqX5iyHJwA7AY8DfwfsLGkFcICk+2ljkJNWZg9KptbzdALMzB4GnjCzR9LrQ/i93Apc6JsBvzKzu1g/A+JSM7sRH33+Efi1mf0WQFIvYAvclX5I2n+nmT2Xrr+hQU7QANRKdMdIGg+8C/hDK2Xuw6ddnsRHtpUBwRx8euVVPANiceCxAhiWvJjgQhyADxKagKeAoyTtnPZvaJATNAA9eiBRpLvlkWiNbjGQCIKOEqILshOiC7IT93SJnuIGyxWuHslL2kn8APMQ3WuQnRBdkJ0QXZCduKcr0B19r43gc60mRFegOz6RaASfazXRvQbZCdEF2QnRBdkJ0QXZ6VIDiWT4qaQkmmxmB0o6EjjOzEalyOGfAG8AfwUWAR8GXjGzWWXVO1iXLtPSSdoEd4i9gqcYWiTpYDwUfWkq9rCZTcZTFAGci0cuh+AaiC4jOmA07q/YDvfQ3ocbrR8vlNlb0jl4UjZw7+vwjHUM2kFXEt0hZvZ1M7sAOCltOx+4vVDmYTObbGaVCbebgD9J+kbOigZt06NDm3qC7zVXeHo1Ea7eCuF7LYfSRSfpsynr8xmdMUwXVmXq8LFBOTTClMkA3Go4HzgxZX6+HHgCtxOehmdwXojnMl0A/CMwDM8YXeEAScuBo4DleFJfAQfiC/ncZ2ZzMnyfYAOU3tKlJLh34qstjcLThb9iZtfSnDD3ATP7Lj4S3SJtewMXXjXz8NUDPpLONxkXdNAglN7SSRqFL+nVD09k+xc8Qe+JwFapWNGAPQQXXHHJsSJFc/VdwNm4kO+pR/2DjtOQo1dJh+FLTGBmV27EeXbAW7vBwJS21jLprmbrRhy9lt7StYSZza3ReZbji/QEDUTp93RBz6Mhu9cy6K6+17LC1btc91oW8QPMQ3SvQXZCdEF2QnRBduKerkBX9b02ore1LUJ0Bbrq5HAjelvbIrrXIDshuiA7IbogOyG6IDudGkhI+jweYrQNMKOSXKSzSJoB3GRmP0+fT0+7eqftfy6UHYg7w14GFpnZoqpzTTOzcRtTn6C+bMzodRae+WZHSf3xXK8r8OQlr9JGxK6k81LZbfGYt11xczSS9gRWmdnVhfKT0/5NgFuqzjXNzMYVxSZpG+A4YA88CvlMPNfsAjNbSlAqG9O9HolbAP8HGIEHTy7Ds99sKGL3vSlO7n2FlE5F/+o7D0GToD8AvISbrFujGNDZB/9uy/GcYmuB60NwjcHGiO42PG/r2Xj26s3xJR8epjlidySwVtI/pYzYFZZJGgs8RxVmtgTYUtLpkibiAnocjyxe3EI9XkhJ6oqm6oE0i7AXsKaS+Toon7qENnUkYrdMuovvtazo4LbIHtrUVSJ2zWwSMAm6bzxdI5JtykTSeyRdmLrNsVX7prVyTIvb076BksZJ+oikAbWub1A/cj57HQtMNbPXkmBOwS2E3wKGSDoaeJB1Pa07SfoKsBcwHpiC53oF+HF6fT/wuqST074HgBOBJfh93u+B/YExZvZq/b9msCFyTg6L5lFp0bs6EB+9zmZ9T+sqM5uBi2nv9Hk6LsJqfovnm+2N54O9AvgznmR4PsldFpRPzpbuSmCipJeArfHplYp3daWkL+Dze0VPa+/U0u2KG6g/mxZGfLSF8/fFkxcPA9aamUl6G1iDp3GPpy8NQhhzEl3Z99rVRq/x6w+yE6ILshPda6Irz9M1Yrh6+F7bSfwA8xDda5CdEF2QneheC3RVC2JXI0RXoBEWvJZkZdej3nWI7jXIToguyE6Irpnzy65AohHqUdc6xORwkJ1o6YLshOgASfukqObLJG1ZUh0OknSWpBskbVtGHVI9Dpd0Zz2vEaJz/gX3SswB/r6MCpjZz83s28CfcOdbdiTtjQfB1tWqGaJrxqpesyPpGGBpif7cT+Fps/ZOAqwLMTnszMZbui0oWBJzktLCHw/cK2kXM3tmQ8fUGjO7KNVlYDLB14UYvQbZie41yE6ILshOiC7IToguyE6ILhOSRkga3Y5yx0vapZPXmCSpzTk+SUMkfaYz568VIbrMSDpW0hmSLpfUR9K3JJ0k6WepyBAze0bSDElfkHRJRUhyLkvvL5O0vaRTJV0paffCNaZVXiVtnspOkHSimT0B7Jf7excJ0WUgZe8elD4ebGaX4Gvu7QW828yuwZfAAF/dFHyFg5vwtVkOljQOX5PlOUkfwheJ3JR1F3+sphe+4sE2eMbw/lXXKIWYHM6Amd0paQT+eOt+SV/DZ/5vBl5MS2fslIq/mV7XAl8BhgITzexOAEl/An4BfIL1F3+s8IakL+Jp6h8DVuKp6R+pukYpxORwyUg6CF80vL+ZTZZ0HLAQOL0eC3ZLGgIMNrPba33udtchRBfkJu7pguyE6ILshOiC7IToguyE6ILshOiC7Pw/+GXmXbthoXEAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 108x216 with 2 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig, (ax1, ax2) = plt.subplots(2, 1, figsize=(1.5,3),\n",
    "                               sharex=True)\n",
    "plt.subplots_adjust(wspace=0.7)\n",
    "\n",
    "meta = pd.read_excel(\n",
    "    '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/gene_lists/aj_kupffer_leptin_6hours_up_kupffer_genes_fc_0.58_pval_0.05_/metascape_result.xlsx',\n",
    "    sheet_name=1)\n",
    "meta_filt = meta.loc[meta['Term'].str.contains('|'.join(leptin_act_terms))]\n",
    "meta_filt = meta_filt.sort_values('LogP', ascending=True)\n",
    "meta_filt = meta_filt.reindex(meta_filt.Term.drop_duplicates().index)\n",
    "\n",
    "y_ind = np.arange(len(leptin_act_terms))\n",
    "enrichment = -meta_filt['LogP']\n",
    "ax1.barh(y_ind,\n",
    "         enrichment,\n",
    "         align='center',\n",
    "         linewidth=1,\n",
    "         edgecolor='k',\n",
    "         color='#bdbdbd')\n",
    "ax1.set_yticks(y_ind)\n",
    "ax1.set_yticklabels(leptin_act_strings)\n",
    "ax1.invert_yaxis()\n",
    "ax1.set_title('Leptin Activated')\n",
    "# ax1.set_xlabel('-log(p-value)')\n",
    "\n",
    "\n",
    "meta = pd.read_excel(\n",
    "    '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/gene_lists/aj_kupffer_control_up_kupffer_genes_fc_0.58_pval_0.05_//metascape_result.xlsx',\n",
    "    sheet_name=1)\n",
    "meta_filt = meta.loc[meta['Term'].str.contains('|'.join(leptin_rep_terms))]\n",
    "meta_filt = meta_filt.sort_values('LogP', ascending=True)\n",
    "meta_filt = meta_filt.reindex(meta_filt.Term.drop_duplicates().index)\n",
    "\n",
    "y_ind = np.arange(len(leptin_rep_terms))\n",
    "enrichment = -meta_filt['LogP']\n",
    "ax2.barh(y_ind,\n",
    "         enrichment,\n",
    "         align='center',\n",
    "         linewidth=1,\n",
    "         edgecolor='k',\n",
    "         color='#bdbdbd')\n",
    "ax2.set_yticks(y_ind)\n",
    "ax2.set_yticklabels(leptin_rep_strings)\n",
    "ax2.invert_yaxis()\n",
    "ax2.set_title('Leptin Repressed')\n",
    "ax2.set_xlabel('-log(p-value)')\n",
    "\n",
    "plt.savefig('../A10_Manuscript_Figures_AllLiverCells/leptin_treatment_GO_long.pdf',\n",
    "            bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Write differential genes to list for metascape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/overall.heatmap.txt',\n",
       " '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/balbc_kupffer_control.vs.balbc_kupffer_leptin_6hours.scatter.homer.txt',\n",
       " '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/balbc_kupffer_leptin_6hours.vs.balbc_kupffer_control.scatter.homer.txt',\n",
       " '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/balbc_kupffer_leptin_6hours.vs.balbc_kupffer_control.scatter.pdf',\n",
       " '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/balbc_kupffer_control.vs.balbc_kupffer_leptin_6hours.scatter.txt',\n",
       " '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/balbc_kupffer_leptin_6hours.vs.balbc_kupffer_control.scatter.txt',\n",
       " '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/aj_kupffer_control.vs.aj_kupffer_leptin_6hours.scatter.homer.txt',\n",
       " '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/balbc_kupffer_control.vs.balbc_kupffer_leptin_6hours.scatter.pdf',\n",
       " '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/aj_kupffer_leptin_6hours.vs.aj_kupffer_control.scatter.txt',\n",
       " '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/aj_kupffer_control.vs.aj_kupffer_leptin_6hours.scatter.txt',\n",
       " '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/aj_kupffer_control.vs.aj_kupffer_leptin_6hours.scatter.pdf',\n",
       " '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/aj_kupffer_leptin_6hours.vs.aj_kupffer_control.scatter.pdf',\n",
       " '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/aj_kupffer_leptin_6hours.vs.aj_kupffer_control.scatter.homer.txt',\n",
       " '../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/overall.heatmap.pdf']"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "glob.glob('../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/*')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['',\n",
       " '',\n",
       " '/K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/aj_kupffer_leptin_6hours',\n",
       " 'vs',\n",
       " 'aj_kupffer_control',\n",
       " 'scatter',\n",
       " 'homer',\n",
       " 'txt']"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "i.split('.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/balbc_kupffer_control.vs.balbc_kupffer_leptin_6hours.scatter.homer.txt\n",
      "ge all transcripts (24940, 18)\n",
      "Diff Gene selected transcripts (9168, 14)\n",
      "ge selected transcripts (9168, 18)\n",
      "../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/balbc_kupffer_leptin_6hours.vs.balbc_kupffer_control.scatter.homer.txt\n",
      "ge all transcripts (24940, 18)\n",
      "Diff Gene selected transcripts (9168, 14)\n",
      "ge selected transcripts (9168, 18)\n",
      "../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/aj_kupffer_control.vs.aj_kupffer_leptin_6hours.scatter.homer.txt\n",
      "ge all transcripts (24940, 18)\n",
      "Diff Gene selected transcripts (9289, 14)\n",
      "ge selected transcripts (9289, 18)\n",
      "../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/aj_kupffer_leptin_6hours.vs.aj_kupffer_control.scatter.homer.txt\n",
      "ge all transcripts (24940, 18)\n",
      "Diff Gene selected transcripts (9289, 14)\n",
      "ge selected transcripts (9289, 18)\n"
     ]
    }
   ],
   "source": [
    "gene_list_dict = {}\n",
    "pval=0.05\n",
    "log2fc=np.log2(1.5)\n",
    "for i in np.sort(glob.glob('../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/c57*c57*homer.txt')):\n",
    "    print(i)\n",
    "    # read in gene expression data\n",
    "    df, tpm, tpm_mat, tpm_mat_quant = import_homer_diffgene(\n",
    "        depath=i,\n",
    "        gepath='../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/expression/HOMER.rawTPM.txt',\n",
    "        gene_name_index=True)\n",
    "\n",
    "    # select differential genes\n",
    "    df = df.loc[~(df.chr.str.contains('chrX|chrY|ChrUn')), :]\n",
    "    sig_genes = df.index[(df['padj'] < pval) & (df['log2FoldChange'] >= log2fc)].to_list()\n",
    "    gene_list_dict[i.split('.')[4]+'_up'] = list(sig_genes)\n",
    "    # print('union set length:', print(len(g_union)))\n",
    "    # print('intersection set length:', print(len(g_intersect)))\n",
    "\n",
    "for i in np.sort(glob.glob('../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/balb*balb*homer.txt')):\n",
    "    print(i)\n",
    "    # read in gene expression data\n",
    "    df, tpm, tpm_mat, tpm_mat_quant = import_homer_diffgene(\n",
    "        depath=i,\n",
    "        gepath='../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/expression/HOMER.rawTPM.txt',\n",
    "        gene_name_index=True)\n",
    "\n",
    "    # select differential genes\n",
    "    df = df.loc[~(df.chr.str.contains('chrX|chrY|ChrUn')), :]\n",
    "    sig_genes = df.index[(df['padj'] < pval) & (df['log2FoldChange'] >= log2fc)].to_list()\n",
    "    gene_list_dict[i.split('.')[4]+'_up'] = list(sig_genes)\n",
    "    # print('union set length:', print(len(g_union)))\n",
    "    # print('intersection set length:', print(len(g_intersect)))\n",
    "    \n",
    "for i in np.sort(glob.glob('../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/differential/aj*aj*homer.txt')):\n",
    "    print(i)\n",
    "    # read in gene expression data\n",
    "    df, tpm, tpm_mat, tpm_mat_quant = import_homer_diffgene(\n",
    "        depath=i,\n",
    "        gepath='../K06b_Strains_Leptin6htoControl_Kupffer_OutliersDropped/expression/HOMER.rawTPM.txt',\n",
    "        gene_name_index=True)\n",
    "\n",
    "    # select differential genes\n",
    "    df = df.loc[~(df.chr.str.contains('chrX|chrY|ChrUn')), :]\n",
    "    sig_genes = df.index[(df['padj'] < pval) & (df['log2FoldChange'] >= log2fc)].to_list()\n",
    "    gene_list_dict[i.split('.')[4]+'_up'] = list(sig_genes)\n",
    "    # print('union set length:', print(len(g_union)))\n",
    "    # print('intersection set length:', print(len(g_intersect)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# for key in gene_list_dict.keys():\n",
    "#     print(key)\n",
    "#     print(len(gene_list_dict[key]), 'genes')\n",
    "#     with open('./gene_lists/'+key+'_kupffer_genes_fc_%.2f_pval_%.2f_.txt' % (log2fc, pval), 'w') as f:\n",
    "#         f.write('Gene\\n')\n",
    "#         for i in gene_list_dict[key]:\n",
    "#             f.write(i+'\\n')\n",
    "#         f.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# look at  overlap with strain specific gene sets"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Read in strain specific gene sets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "strain_spec_kc_genes = {}\n",
    "for gene_list_path in glob.glob('/home/h1bennet/strains_rna/results/K00_Strains_Control_Kupffer_RNA/gene_lists/*txt'):\n",
    "    gene_list = []\n",
    "    gene_list_name = gene_list_path.split('/')[-1].split('_kupffer')[0]\n",
    "    with open(gene_list_path, 'r') as f:\n",
    "        strain_spec_kc_genes[gene_list_name] = [i.strip() for i in f.readlines()][1:]\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There is mild overlap between these gene sets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['balbc_kupffer_leptin_6hours_up', 'balbc_kupffer_control_up', 'aj_kupffer_leptin_6hours_up', 'aj_kupffer_control_up'])"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_list_dict.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "aj_intersection\n",
      "0\n",
      "set()\n",
      "balb_intersection\n",
      "3\n",
      "{'Lgals1', 'Ifi27l2a', 'Trem2'}\n",
      "c57_intersection\n",
      "0\n",
      "set()\n",
      "balb_union\n",
      "6\n",
      "{'Lgals1', 'Ifi27l2a', 'Trem2', 'Per1', 'Anxa2', 'Rcan1'}\n",
      "aj_union\n",
      "3\n",
      "{'Marco', 'Lgals1', 'Ikbke'}\n",
      "c57_union\n",
      "6\n",
      "{'Ikbke', 'Per1', 'Marco', 'Rgs1', 'D17H6S56E-5', 'Mki67'}\n"
     ]
    }
   ],
   "source": [
    "for key in strain_spec_kc_genes.keys():\n",
    "    print(key)\n",
    "    print(len(set(gene_list_dict['aj_kupffer_leptin_6hours_up']).intersection(strain_spec_kc_genes[key])))\n",
    "    print(set(gene_list_dict['aj_kupffer_leptin_6hours_up']).intersection(strain_spec_kc_genes[key]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "230\n"
     ]
    }
   ],
   "source": [
    "print(len(strain_spec_kc_genes['balb_union']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27\n"
     ]
    }
   ],
   "source": [
    "print(len(set(gene_list_dict['aj_kupffer_control_up'])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "aj_intersection\n",
      "0\n",
      "set()\n",
      "balb_intersection\n",
      "0\n",
      "set()\n",
      "c57_intersection\n",
      "0\n",
      "set()\n",
      "balb_union\n",
      "2\n",
      "{'Lepr', 'Rnf144b'}\n",
      "aj_union\n",
      "4\n",
      "{'Fosb', 'Mturn', 'Rnf144b', 'Vipr1'}\n",
      "c57_union\n",
      "4\n",
      "{'Fosb', 'Ncoa1', 'Lepr', 'Vipr1'}\n"
     ]
    }
   ],
   "source": [
    "for key in strain_spec_kc_genes.keys():\n",
    "    print(key)\n",
    "    print(len(set(gene_list_dict['aj_kupffer_control_up']).intersection(strain_spec_kc_genes[key])))\n",
    "    print(set(gene_list_dict['aj_kupffer_control_up']).intersection(strain_spec_kc_genes[key]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "227"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(strain_spec_kc_genes['aj_union'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27\n"
     ]
    }
   ],
   "source": [
    "print(len(set(gene_list_dict['aj_kupffer_control_up'])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(7.043088321310197, 0.0039301583384493705)"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats.fisher_exact([[9289-227-27-4, 27-4], [227-4, 4]], alternative='greater')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "import scipy.stats as stats"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## test enrichment of balb union genes in leptin induced set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(9289, 19)"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "de.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9289\n",
      "230\n",
      "37\n",
      "6\n"
     ]
    }
   ],
   "source": [
    "print(de.shape[0])\n",
    "print(len(set(strain_spec_kc_genes['balb_union'])))\n",
    "print(len(set(gene_list_dict['aj_kupffer_leptin_6hours_up'])))\n",
    "print(len(set(gene_list_dict['aj_kupffer_leptin_6hours_up']).intersection(strain_spec_kc_genes['balb_union'])))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Contingency table explained:  \n",
    "\n",
    "genes in dataset: 9289  \n",
    "genes 2 fold up in balb union: 230  \n",
    "genes 1.5 fold induced by leptin in aj: 37  \n",
    "overlap between two gene sets: 6.  \n",
    "\n",
    "|  |Not in A              | in A |\n",
    "|:---:|:----------------------:|:-----:\n",
    "Not in B|9289 - 224 - 31 - 6 | 37-6\n",
    "In B|230-6                | 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(7.8058755760368665, 0.0002628836657345142)"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats.fisher_exact([[9289-224-31, 37-6], [230-6, 6]], alternative='greater')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9289\n",
      "230\n",
      "27\n",
      "2\n"
     ]
    }
   ],
   "source": [
    "print(de.shape[0])\n",
    "print(len(set(strain_spec_kc_genes['balb_union'])))\n",
    "print(len(set(gene_list_dict['aj_kupffer_control_up'])))\n",
    "print(len(set(gene_list_dict['aj_kupffer_control_up']).intersection(strain_spec_kc_genes['balb_union'])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3.170526315789474, 0.14325594424333038)"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats.fisher_exact([[9289-224-27-2, 27-2], [230-2, 2]], alternative='greater')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Lets look at overlap between BALB and AJ differential genes in particular"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/ttroutman/strainsKupffer/rnaKupfferNASH/rnaDiff/aj_control.vs.balbcj_control.scatter.txt\n"
     ]
    }
   ],
   "source": [
    "pval = 0.05\n",
    "log2fc = np.log2(1.5)\n",
    "for i in np.sort(glob.glob('/home/ttroutman/strainsKupffer/rnaKupfferNASH/rnaDiff/aj_control.vs.balbcj_control.scatter.txt')):\n",
    "    print(i)\n",
    "    # read in gene expression data\n",
    "    df = pd.read_csv(i,\n",
    "                 sep='\\t',\n",
    "                 index_col=0)\n",
    "    \n",
    "    # select differential genes\n",
    "    # df = df.loc[~(df.chr.str.contains('chrX|chrY|ChrUn')), :]\n",
    "    sig_genes = df.index[(df['padj'] < pval) & (df['log2FoldChange'] >= log2fc)].to_list()\n",
    "    gene_list_dict[i.split('/')[-1]] = list(sig_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(7619, 7)"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "import scipy.stats as stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "aj_control.vs.balbcj_control.scatter.txt\n",
      "10\n",
      "{'Hspa1a', 'Lgals1', 'Ifi27l2a', 'Trem2', 'Rpl37rt', 'Sem1', 'Per1', 'Anxa3', 'Fabp7', 'Rcan1'}\n"
     ]
    }
   ],
   "source": [
    "key='aj_control.vs.balbcj_control.scatter.txt'\n",
    "print(key)\n",
    "print(len(set(gene_list_dict['aj_kupffer_leptin_6hours_up']).intersection(gene_list_dict[key])))\n",
    "print(set(gene_list_dict['aj_kupffer_leptin_6hours_up']).intersection(gene_list_dict[key]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "aj_control.vs.balbcj_control.scatter.txt\n",
      "1\n",
      "{'Lepr'}\n"
     ]
    }
   ],
   "source": [
    "key='aj_control.vs.balbcj_control.scatter.txt'\n",
    "print(key)\n",
    "print(len(set(gene_list_dict['aj_kupffer_control_up']).intersection(gene_list_dict[key])))\n",
    "print(set(gene_list_dict['aj_kupffer_control_up']).intersection(gene_list_dict[key]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "239\n"
     ]
    }
   ],
   "source": [
    "print(len(gene_list_dict[key]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## test enrichment of balb up vs aj genes in leptin induced set"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Contingency table explained:\n",
    "genes in dataset: 7619\n",
    "genes 2 fold up in balb vs aj: 239\n",
    "genes 1.5 fold induced by leptin in aj: 37\n",
    "overlap between two gene sets: 10.\n",
    "\n",
    "|  |Not in A              | in A |\n",
    "|:---:|:----------------------:|:-----:\n",
    "Not in B|7353 - 229 - 27 - 10 | 37-10\n",
    "In B|229-10                | 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(11.985455775410113, 1.1897668481120654e-07)"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats.fisher_exact([[7353-229-27-10, 37-10], [229-10, 10]], alternative='greater')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0.8645224171539961, 0.6903356597654796)"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats.fisher_exact([[7353-229-27-1, 37-1], [229-1, 1]], alternative='greater')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "dataviz",
   "language": "python",
   "name": "dataviz"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
